internal/command/testdata/test/complex_condition/main.tftest.hcl

```diff
@@ -0,0 +1,52 @@
+
+variables {
+  foo = [{
+    bar = "baz",
+    qux = "quux",
+  }]
+}
+
+run "validate_diff_types" {
+  variables {
+    tr1 = {
+    "iops" = tonumber(null)
+    "size" = 60
+}
+    tr2 = {
+    iops = null
+    size = 60
+}
+  }
+  assert {
+    condition = var.tr1 == var.tr2 
+    error_message = "expected to fail"
+  }
+}
+
+run "validate_output" {
+  assert {
+    condition = output.foo == var.foo[0]
+    error_message = "expected to fail"
+  }
+}
+
+run "validate_complex_output" {
+  assert {
+    condition = output.complex == var.foo
+    error_message = "expected to fail"
+  }
+}
+
+run "validate_complex_output_sensitive" {
+  assert {
+    condition = output.complex == output.complex_sensitive
```

Comment:
What would the output look like if the binary expression included nested expressions?
E.g. `(output.complex == output.complex_sensitive) == true`?

Would the output contain all the information, or summarise the left side expression?

Reply:
Good question!. The LHS would still be `(output.complex == output.complex_sensitive)`, with it being a type of `ParenthesesExpr`. This is HCL implementation detail, but it feels like a good example to document. Therefore, in the diff, the LHS may simply be something like `true`. 

Nevertheless, in addition to the LHS/RHS and diff, we also display the individual references, e.g

```
  28:     condition = output.foo == var.foo[0]
    ├────────────────
...
    │ Diff:
...
    │ output.foo is {
    │     "bar": "notbaz",
    │     "qux": "quux"
    │   }
    │ var.foo[0] is {
    │     "bar": "baz",
    │     "qux": "quux"
    │   }
```

I intend to move this one to the verbose mode though, so we keep the default mode focused on just the LHS and RHS values.